### PR TITLE
Add distribution tracker to public channel even  if distribution Id is not stored

### PR DESCRIPTION
**Pull request by meiravf**
Distribution tracker should always be added to the group channel even if the distribution group Id is empty. 

If not doing so when installing the app from scratch (removing the previous version or installing on the first time) the stored  distribution Id is empty which results in not adding the tracker into the public channels . Afterwards the browser is opened and we retrieve the distribution Id but its too late the tracker is not part of  the the public channels and the group Id is not logged.

### CODE REVIEW

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m [hunk 1]
```diff
@@ -108,7 +108,8 @@ - (NSString *)groupId {
 
 - (MSInitializationPriority)initializationPriority {
 
-  // Initialize Distribute before Analytics to add distributionGroupId field to the first startSession event after app start.
+  // Initialize Distribute before Analytics to add distributionGroupId field to the first startSession event after app
```
Comment (jaelim-ms) [inline AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: Can you make it single line? Or use /* */ for multiline comment.

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m [hunk 2]
```diff
@@ -125,10 +126,15 @@ - (void)applyEnabledState:(BOOL)isEnabled {
...
 
     // Enable the distribute info tracker.
-    NSString *distributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
-    [self.distributeInfoTracker updateDistributionGroupId:distributionGroupId];
     [self.channelGroup addDelegate:self.distributeInfoTracker];
-  } else {
+
+    // Update distribute info tracker with stored distributionGroupId
+    NSString *distributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
+    if (distributionGroupId) {
```
Comment (jaelim-ms) [inline AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: Please reformat this if statement. Hard to read it.

Comment (TroubleMakerBen) [reply AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: I agree.

**Comment by meiravf** [comment]
@jaelim-ms I have updated the title of the PR, you are right the `term` enable is wrong .
Its actually adding the `distributeInfoTracker` to the `publicChannel`.

**Comment by meiravf** [comment]
@jaelim-ms Yes we have done e2e on iOS, I thought that macOS is not supported with in-app updates.

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m [hunk 3]
```diff
@@ -125,10 +125,15 @@ - (void)applyEnabledState:(BOOL)isEnabled {
...
 
     // Enable the distribute info tracker.
-    NSString *distributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
-    [self.distributeInfoTracker updateDistributionGroupId:distributionGroupId];
     [self.channelGroup addDelegate:self.distributeInfoTracker];
-  } else {
+
+    // Store distributionGroupId in distributeInfoTracker
+    NSString *distributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
+    if (distributionGroupId) {
```
Comment (jaelim-ms) [inline AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: Please format this again. The indentation for `}` seems wrong in this `if` statement.

**Comment by meiravf** [comment]
@jaelim-ms done I have re-formatted the file.

**Comment by meiravf** [comment]
@jaelim-ms I see that the formatting didn't work well. fixing..

**Pull request merged**